libpod/container_internal.go

```diff
@@ -1082,8 +1082,11 @@ func (c *Container) addLocalVolumes(ctx context.Context, g *generate.Generator,
 		if MountExists(g.Mounts(), k) {
 			continue
 		}
-		volumePath := filepath.Join(c.config.StaticDir, "volumes", k)
-
+		volume, err := c.runtime.NewVolume(ctx, []VolumeCreateOption{}...)
+		if err != nil {
+			return errors.Wrapf(ErrCtrStateInvalid, "failed to create volume for %s in %s for container %s", k, mountPoint, c.ID())
+		}
+		volumePath := filepath.Join(volume.config.MountPoint, k)
```

Comment:
Nuke the filepath.Join() - why do we need it? There's only one thing in the volume, no need to make a subdir in it

Reply:
The k in this case is the directory that is being volume mounted in.

In my tests It happens to be /myvol.  I believe this is necessary.

Reply:
Shouldn't be - we're bind-mounting it into the given path in the container, no need to have the volume path appended again after the path for the named volume

Reply:
Right, we have already created a new volume with a generated name, which will be bind-mounted to /myvol inside the container. So no need to do the filepath.Join() and create a sub-dir inside that.

Reply:
Fixed